src/core/socket-bind.c

```diff
@@ -178,3 +188,20 @@ int socket_bind_install(Unit *u) {
 }
 
 #endif
+int socket_bind_restore(Unit *u, int fd) {
```

Comment:
@keszybz 

Renamed this helper as `socket_bind_add_initial_link_fd`, hope it's more descriptive.

This doesn't saves a socket: this stores deserialized fds of bpf links existing before daemon-re{load|exec}, hence "initial" in the name. 
That resolves a problem described in https://github.com/systemd/systemd/pull/17495#issue-513195381, TL;DR: unit detaching bpf -> re{load|exec} -> attaching bpf is a security vulnerability because either device or a port or another resource may be "stolen" while daemon is reinitializing.

These fds are stored up to the point when it is save to close them, i.e. after new links are created. Closing link fd decrements bpf program refcount, for a sole link that means detaching a program from cgroup and unloading from kernel.